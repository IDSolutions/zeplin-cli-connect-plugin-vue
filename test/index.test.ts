import Plugin from "../src";

describe("Connected Components Vue Plugin", () => {
    test("MyButton.vue snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/MyButton.vue",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("MyIcon.vue snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/MyIcon.vue",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });
});
