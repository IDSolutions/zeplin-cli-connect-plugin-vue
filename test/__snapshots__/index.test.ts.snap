// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Components Vue Plugin MyButton.vue snippet creation 1`] = `
Object {
  "description": "General component description",
  "lang": "html",
  "snippet": "
<MyButton
  color=\\"string\\"
  disabled=\\"boolean\\"
  href=\\"string\\"
  icon=\\"string\\"
  iconPlacement=\\"string\\"
  iconProps=\\"object\\"
  label=\\"string\\"
  size=\\"string\\"
  to=\\"string|object\\"
/>
        ",
}
`;

exports[`Connected Components Vue Plugin MyIcon.vue snippet creation 1`] = `
Object {
  "description": "General component description",
  "lang": "html",
  "snippet": "
<MyIcon
  name=\\"string\\"
  width=\\"number\\"
  height=\\"number\\"
  title=\\"string\\"
  solid=\\"boolean\\"
/>
        ",
}
`;
